specs/agents/metadata.md

```diff
@@ -99,6 +99,111 @@ metadata is available.
+
+From the returned metadata, the following fields are useful
+
+| Cloud metadata field  | AWS Metadata field  |
+| --------------------  | ------------------- |
+| `account.id`          | `accountId`         |
+| `instance.id`         | `instanceId`        |
+| `availability_zone`   | `availabilityZone`  |
+| `machine.type`        | `instanceType`      |
+| `provider`            | aws                 |
+| `region`              | `region`            |
+
+#### GCP metadata
+
+Metadata about a GCP machine instance can be retrieved from the 
+metadata service, `http://metadata.google.internal`.
+
+An example with curl
+
+```sh
+curl -X GET "http://metadata.google.internal/computeMetadata/v1/?recursive=true" -H "Metadata-Flavor: Google"
+```
+
+From the returned metadata, the following fields are useful
+
+| Cloud metadata field  | AWS Metadata field  |
+| --------------------  | ------------------- |
+| `instance.id`         | `instance.id`       |
+| `instance.name`       | `instance.name`     |
+| `project.id`          | `project.numericProjectId` as a string |
+| `project.name`        | `project.projectId` |
+| `availability_zone`   | last part of `instance.zone`, split by `/`  |
+| `machine.type`        | last part of `instance.machineType`, split by `/` |
+| `provider`            | gcp                 |
+| `region`              | last part of `instance.zone`, split by `-`            |
+
+#### Azure metadata
+
+##### Azure VMs
+
+Metadata about an Azure VM can be retrieved from the internal metadata
+endpoint, `http://169.254.169.254`.
+
+An example with curl
+
+```sh
+curl -X GET "http://169.254.169.254/metadata/instance/compute?api-version=2019-08-15" -H "Metadata: true"
+```
+
+From the returned metadata, the following fields are useful
+
+| Cloud metadata field  | AWS Metadata field  |
+| --------------------  | ------------------- |
+| `account.id`          | `subscriptionId`    |
+| `instance.id`         | `vmId`              |
+| `instance.name`       | `name`              |
+| `project.name`        | `resourceGroupName` |
+| `availability_zone`   | `zone`              |
+| `machine.type`        | `vmSize`            |
+| `provider`            | azure               |
+| `region`              | `location`          |
+
+##### Azure App Services _(Optional)_
+
+Azure App Services are a PaaS offering within Azure which does not
+have access to the internal metadata endpoint. Metadata about
+an App Service can however be retrieved from environment variables
+
+
+| Cloud metadata field  | Environment variable |
+| --------------------  | ------------------- |
+| `account.id`          | first part of `WEBSITE_OWNER_NAME`, split by `+` |
+| `instance.id`         | `WEBSITE_INSTANCE_ID` |
+| `instance.name`       | `WEBSITE_SITE_NAME` |
+| `project.name`        | `WEBSITE_RESOURCE_GROUP` |
+| `provider`            | azure               |
+| `region`              | last part of `WEBSITE_OWNER_NAME`, split by `-`, trim end `"webspace"`          |
+
```

Comment:
@russcam datapoint: When I deployed a node application via azure app services my region looked like
```
    "WEBSITE_RESOURCE_GROUP": "appsvc_linux_centralus",
```

i.e. the `WEBSITE_OWNER_NAME` was delimited with underscores characters (`_`) and not the hyphen/dash.  You can see [this here](https://elastic-nodejs-agent-metadata.azurewebsites.net/show-env) (will be slow in coming up if the app's gone idle)

What do you think about spec-ing this as splitting by `-` OR `_` -- or perhaps even "splitting by non-alpha-numeric-characters"?

Reply:
Thanks for the additional data point. Looks like there may be some cross-language and cross-platform differences in environment variable values. For example, the node app value is

```
"WEBSITE_OWNER_NAME": "7657426d-c4c3-44ac-88a2-3b2cd59e6dba+appsvc_linux_centralus-CentralUSwebspace-Linux"
```

The extracted region should be `CentralUS`, though the value includes a `-Linux` suffix which wasn't present on the .NET app service.

I think what might be good for me to do before updating the PR is to deploy some other language app services to see if there is more variation in values. It might be a simple case of updating the `WEBSITE_OWNER_NAME` rule to substring from 0 up to start of `webspace`, then split by `-` and take the last value.